libraries/ESP8266mDNS/examples/LEAmDNS/mDNS_Clock/mDNS_Clock.ino

```diff
@@ -0,0 +1,329 @@
+        }
+      }
+    }
+  }
+  return true;
+}
+
+
+/*
+ * handleHTTPClient
+ */
+void handleHTTPClient(WiFiClient& client) {
+  Serial.println("");
+  Serial.println("New client");
+
+  // Wait for data from client to become available
+  while (client.connected() && !client.available()) {
+    delay(1);
+  }
+
+  // Read the first line of HTTP request
+  String req = client.readStringUntil('\r');
+
+  // First line of HTTP request looks like "GET /path HTTP/1.1"
+  // Retrieve the "/path" part by finding the spaces
+  int addr_start = req.indexOf(' ');
+  int addr_end = req.indexOf(' ', addr_start + 1);
+  if (addr_start == -1 || addr_end == -1) {
+    Serial.print("Invalid request: ");
+    Serial.println(req);
+    return;
+  }
+  req = req.substring(addr_start + 1, addr_end);
+  Serial.print("Request: ");
+  Serial.println(req);
+  client.flush();
+
+  // Get current time
+  time_t now = time(nullptr);;
+  struct tm timeinfo;
+  gmtime_r(&now, &timeinfo);
+ 
+  String s;
+  if (req == "/") {
+    IPAddress ip = WiFi.localIP();
+    String ipStr = String(ip[0]) + '.' + String(ip[1]) + '.' + String(ip[2]) + '.' + String(ip[3]);
+    s = "HTTP/1.1 200 OK\r\nContent-Type: text/html\r\n\r\n<!DOCTYPE HTML>\r\n<html>Hello from ESP8266 at ";
+    s += ipStr;
+    // Simple addition of the current time
+    s += "\r\nCurrent time is: ";
+    s += getTimeString();
+    // done :-)
+    s += "</html>\r\n\r\n";
+    Serial.println("Sending 200");
+  } else {
+    s = "HTTP/1.1 404 Not Found\r\n\r\n";
+    Serial.println("Sending 404");
+  }
+  client.print(s);
+
+  Serial.println("Done with client");
+}
+
+
+/*
+ * setup
+ */
+void setup(void) {
+  Serial.begin(115200);
+  
+  // Connect to WiFi network
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, password);
+  Serial.println("");
+
+  // Wait for connection
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("");
+  Serial.print("Connected to ");
+  Serial.println(ssid);
+  Serial.print("IP address: ");
+  Serial.println(WiFi.localIP());
+  
+  // Sync clock
+  setClock();
+  
+  // Setup MDNS responder
+  LEAmDNS::MDNS.setProbeResultCallback(MDNSProbeResultCallback, 0);
+  // Init the (currently empty) host domain string with 'esp8266'
+  if ((!LEAmDNS::MDNSResponder::indexDomain(pcHostDomain, 0, "esp8266")) ||
+      (!LEAmDNS::MDNS.begin(pcHostDomain))) {
+    Serial.println("Error setting up MDNS responder!");
+    while (1) { // STOP
+      delay(1000);
+    }
+  }
+  Serial.println("MDNS responder started");
+
+  // Start TCP (HTTP) server
+  server.begin();
+  Serial.println("TCP server started");
+}
+
+
+/*
+ * loop
+ */
+void loop(void) {
+  // Check if a client has connected
+  WiFiClient    client = server.available();
+  if (client) {
+    handleHTTPClient(client);
+  }
+  
+  // Allow MDNS processing
+  LEAmDNS::MDNS.update();
+  
+  // Update time (if needed)
+  static    unsigned long ulNextTimeUpdate = UPDATE_CYCLE;

```

Comment:
Same issue w/millis wraparound here, I think.

Reply:
I‘ve only now learned, that a millis() roll over happens every 72 minutes... Is there any alternative way? Maybe some tick counter??

Reply:
I’m getting confused ... is the millis() rollover really hard-connected to the micros() rollover?? Otherwise it would be a rollover time of about 6 weeks... (haven’t seen an ESP running that long without running out of memory and needs a restart :-( )

Reply:
@LaborEtArs millis() does a rollover, so you can't compare the value returned by millis() against a delta. Instead, you have to compare a delta against a value. A delta (i.e.: a millis() - op_start_time) considers the case of rollover, so the result is always correct if you happen to be caught in the middle of a rollover. 
Well, "always correct" as long as you don't have 2 rollovers in between 😛 .
This kind of bug, and having to think your way through it every time you need to do a timeout operation, is the motivation behind #5198 : have the details abstracted and simplify coding.
But it's not merged yet.

Reply:
Ok, got the problem!
I guess, I will incorporate a similar class as your pooledTimeout here.
BTW: There had been a short discussion about the types and type names in that thread. Personally I really don’t like the use of ‘using ... unsigned int’ for standard types, as it extremely reduces the readability of the code. Of course not, if there are just one or two of those ‘using’ declarations, but the more the worse it becomes. I would plead for standard types like uint32_t in cases like this.

Reply:
Ok, got the problem!
I guess, I will incorporate a similar class as your pooledTimeout here.
BTW: There had been a short discussion about the types and type names in that thread. Personally I really don’t like the use of ‘using ... unsigned int’ for standard types, as it extremely reduces the readability of the code. Of course not, if there are just one or two of those ‘using’ declarations, but the more the worse it becomes. I would plead for standard types like uint32_t in cases like this.